### DISCUSSION TITLE
Device stop sending readings after changing Hostname or MQTT configs

### DISCUSSION THREAD

**Discussion by miky-boy** [arendst/Tasmota#15417] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

Hi, after my problem yesterday when i updated the version from 8.1 to 11.1.0 i realized some problems now which i detail below.

I think i use the old version of ZMAI-90, so the TuyaMCU commands are not working. I'm not sure if anyone could confirm but after my long tests, seems to be.

But what i realized now is, if i configure the sensor to send readings it will send as suposed, but if i change the hostname or the IP address or even the MQTT parameters it stop sending the readings. Still appear the correct command from serial as below .. but nothing.

16:35:40.711 CMD: Grp 0, Cmd 'SERIALSEND', Idx 5, Len 16, Pld -99, Data 'fe010f080000001c'
16:35:40.716 MQT: stat/SensorQuadro/RESULT = {"SerialSend":"Done"}

Even if i reboot.. nothing.

After a long battle of tests, if i change the Hostname, configure the MQTT and change the IP and only AFTER, i configure the reading sensor, it works! It means, the readings should be the last thing to config. 

I leave below the commands i use to configure the Readings.

Could you help me please?

#Configure the serial port to the correct speed and in the Tasmota console type: 
Baudrate 9600

#Configure the serial port to the correct mode and in the Tasmota console type: 
SerialConfig 8E1  

#You will need to create a rule to periodically send a command to the V9821, to obtain the measurements and in the Tasmota console type:
Rule1 on System#Boot do RuleTimer1 5 endon on Rules#Timer=1 do backlog Baudrate 9600; SerialSend5 fe010f080000001c; RuleTimer1 5 endon

#Next, enable the rule by typing:
Rule1 on

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
16:39:29.155 CMD: Backlog Template; Module; GPIO 255
16:39:29.157 SRC: WebConsole from 192.168.2.41
16:39:29.160 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 26, Pld -99, Data 'Template; Module; GPIO 255'
16:39:29.238 SRC: Backlog
16:39:29.241 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
16:39:29.247 MQT: stat/SensorQuadro/RESULT = {"NAME":"ZMAi-90","GPIO":[0,3200,0,3232,0,0,0,0,224,64,0,0,0,0],"FLAG":0,"BASE":18}
16:39:29.462 SRC: Backlog
16:39:29.465 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
16:39:29.470 MQT: stat/SensorQuadro/RESULT = {"Module":{"0":"ZMAi-90"}}
16:39:29.711 SRC: Backlog
16:39:29.714 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data '255'
16:39:29.724 MQT: stat/SensorQuadro/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"3200":"Serial Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3232":"Serial Rx"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"64":"Button_n1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
16:39:34.389 RUL: RULES#TIMER=1 performs "backlog Baudrate 9600; SerialSend5 fe010f080000001c; RuleTimer1 5"

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
16:40:08.312 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 19, Pld -99, Data 'Rule1; Rule2; Rule3'
16:40:08.420 SRC: Backlog
16:40:08.424 CMD: Grp 0, Cmd 'RULE', Idx 1, Len 0, Pld -99, Data ''
16:40:08.430 MQT: stat/SensorQuadro/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":128,"Free":383,"Rules":"on System#Boot do RuleTimer1 5 endon on Rules#Timer=1 do backlog Baudrate 9600; SerialSend5 fe010f080000001c; RuleTimer1 5 endon"}}
16:40:08.653 SRC: Backlog
16:40:08.656 CMD: Grp 0, Cmd 'RULE', Idx 2, Len 0, Pld -99, Data ''
16:40:08.663 MQT: stat/SensorQuadro/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
16:40:08.904 SRC: Backlog
16:40:08.907 CMD: Grp 0, Cmd 'RULE', Idx 3, Len 0, Pld -99, Data ''
16:40:08.913 MQT: stat/SensorQuadro/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
16:40:10.424 RUL: RULES#TIMER=1 performs "backlog Baudrate 9600; SerialSend5 fe010f080000001c; RuleTimer1 5"

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
16:40:56.431 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 1, Pld 0, Data '0'
16:40:56.437 MQT: stat/SensorQuadro/STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"SensorQuadro","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
16:40:56.452 MQT: stat/SensorQuadro/STATUS1 = {"StatusPRM":{"Baudrate":9600,"SerialConfig":"8E1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:19:39","StartupUTC":"2022-04-21T15:21:17","Sleep":50,"CfgHolder":4617,"BootCount":8,"BCResetTime":"2022-04-21T15:57:17","SaveCount":28,"SaveAddress":"F8000"}}
16:40:56.465 MQT: stat/SensorQuadro/STATUS2 = {"StatusFWR":{"Version":"11.1.0(tasmota)","BuildDateTime":"2022-04-13T06:40:42","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"377/699"}}
16:40:56.479 MQT: stat/SensorQuadro/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":4,"MqttLog":2,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Tity",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00C08009","2805C80001000600003C5A0A190000000000","00000088","00006000","00004000"]}}
16:40:56.499 MQT: stat/SensorQuadro/STATUS4 = {"StatusMEM":{"ProgramSize":626,"Free":376,"Heap":21,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"1540C8","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","04000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,56,62","Sensors":"1,2,3,4,5,6"}}
16:40:56.512 MQT: stat/SensorQuadro/STATUS5 = {"StatusNET":{"Hostname":"SensorQuadro","IPAddress":"192.168.2.201","Gateway":"192.168.2.250","Subnetmask":"255.255.255.0","DNSServer1":"192.168.2.250","DNSServer2":"0.0.0.0","Mac":"D8:BF:C0:DE:36:E9","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
16:40:56.525 MQT: stat/SensorQuadro/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.2.17","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_DE36E9","MqttUser":"MQTT_Server_Username","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
16:40:56.540 MQT: stat/SensorQuadro/STATUS7 = {"StatusTIM":{"UTC":"2022-04-21T15:40:56","Local":"2022-04-21T16:40:56","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"05:47","Sunset":"19:49"}}
16:40:56.553 MQT: stat/SensorQuadro/STATUS10 = {"StatusSNS":{"Time":"2022-04-21T16:40:56"}}
16:40:56.566 MQT: stat/SensorQuadro/STATUS11 = {"StatusSTS":{"Time":"2022-04-21T16:40:56","Uptime":"0T00:19:39","UptimeSec":1179,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"Tity","BSSId":"A0:3E:6B:A5:A4:D9","Channel":12,"Mode":"11n","RSSI":70,"Signal":-65,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
16:41:46.401 CMD: Grp 0, Cmd 'BAUDRATE', Idx 1, Len 4, Pld 9600, Data '9600'
16:41:46.406 MQT: stat/SensorQuadro/RESULT = {"Baudrate":9600}
16:41:46.648 SRC: Backlog
16:41:46.651 CMD: Grp 0, Cmd 'SERIALSEND', Idx 5, Len 16, Pld -99, Data 'fe010f080000001c'
16:41:46.658 MQT: stat/SensorQuadro/RESULT = {"SerialSend":"Done"}
16:41:46.900 SRC: Backlog
16:41:46.903 CMD: Grp 0, Cmd 'RULETIMER', Idx 1, Len 1, Pld 5, Data '5'
16:41:46.910 MQT: stat/SensorQuadro/RESULT = {"T1":5,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}

```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
Should appear the RESULT from the Reading sensor

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by miky-boy** [comment]
Hi,

The correct output should be:

17:24:33.936 CMD: Grp 0, Cmd 'RULETIMER', Idx 1, Len 1, Pld 5, Data '5'
17:24:33.944 MQT: stat/SensorQuadro/RESULT = {"T1":4,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:24:33.959 MQT: tele/SensorQuadro/RESULT = {"SerialReceived":"FE01087750840007240000836402000050000008820500482202003938060013090000EE"}

**Comment by miky-boy** [comment]
Hi,

You moved this message to discussion, but for me seems to be a bug. Probably my explanation before was not clear but let me try to describe my issue on other way:

- A) If i configure the device with the following order:
1. Set the Hostname
2. Set the IP Address
3. Set the MQTT 
4. Set the Serial to get Readings
-> Is working properly and i get results from the serial like the message below:

17:24:33.936 CMD: Grp 0, Cmd 'RULETIMER', Idx 1, Len 1, Pld 5, Data '5'
17:24:33.944 MQT: stat/SensorQuadro/RESULT = {"T1":4,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
**17:24:33.959 MQT: tele/SensorQuadro/RESULT = {"SerialReceived":"FE01087750840007240000836402000050000008820500482202003938060013090000EE"}**

- B) But if after i change again some setting, it will stop send the readings from serial and i get the message below without any SerialReceived.

16:41:46.903 CMD: Grp 0, Cmd 'RULETIMER', Idx 1, Len 1, Pld 5, Data '5'
16:41:46.910 MQT: stat/SensorQuadro/RESULT = {"T1":4,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}

I don't see any reason for that. Could you help me please?
Thanks and Regards

**Reply by barbudor** [reply]
You rule do not match the one from https://templates.blakadder.com/ZMAi-90.html
You shouldn't have a `Baudrate 9600` in the `on Rules#Timer=1` rule. Changing the baudrate continuously could be source for problems.

The data string end is also different but I can't advise on that.

What would be interresting are the logs when the timer expires and the rule is triggered.
Do you see the rule being triggered and the SerialSend performed ?

**Comment by miky-boy** [comment]
Hi my friend,

I always used since the beguining this config. Is related with an old issue -> https://github.com/arendst/Tasmota/issues/7496

According the log seems to be triggered. I made a new test now and after entering the command "Hostname SensorQuadro", the device reboot and stop the readings :(. I got this log below:

19:23:34.414 RUL: RULES#TIMER=1 performs "backlog Baudrate 9600; SerialSend5 fe010f080000001c; RuleTimer1 5"
19:23:34.450 RSL: RESULT = {"Baudrate":9600}
19:23:34.700 RSL: RESULT = {"SerialSend":"Done"}
19:23:34.951 RSL: RESULT = {"T1":5,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}

**Comment by miky-boy** [comment, marked as answer]
Hi,
I made other test now and reverted to the old version [Tasmota 8.5.1 by Theo Arends](https://bit.ly/tasmota) and this issue doesn't happen.
As i said on my first message, seems to be any bug or incompatibility with my ZMAI-90 template.
If you have any suggestion to fix on the last release 11.1 please tell me.
Thanks and Regards

**Reply by barbudor** [reply]
The issue you linked has nothing to do with the ZMAI

Start by fixing your rule and remove the `Baudrate 9600`command from the ` On Rules#Tmer=1 DO...`

**Reply by miky-boy** [reply]
Hi,
About the issue, well i had it before on my ZMAI device. If was related with the relase version or other bug i don't know. Honestly i configured the ZMAI device on that time (almost 2 years ago) and never touched anymore.
Only now to try to change my approch on readings i decided to upgrade to version 11,1 following the upgrade path.
Honestly i don't see any relation with this Baudrate after change any attribute on a configured device.
Thanks and Regards,

**Reply by miky-boy** [reply]
Hi,

I made other test now on my version Tasmota 8.5.1 where is working fine any edit on attributes after the reading is working. I was testing your sugestion to remove the Baudrate from rule, but i detected after a device reboot is not sending the readings again. I need to execute Baudrate 9600 and the readings start again. Any solution to make it persistent?

My rules now are:

#Configure the serial port to the correct speed and in the Tasmota console type: 
Baudrate 9600

#Configure the serial port to the correct mode and in the Tasmota console type: 
SerialConfig 8E1

#You will need to create a rule to periodically send a command to the V9821, to obtain the measurements and in the Tasmota console type:
Rule1 on System#Boot do RuleTimer1 10 endon on Rules#Timer=1 do backlog SerialSend5 fe010f080000001c; RuleTimer1 10 endon
        	
#Next, enable the rule by typing:
Rule1 1

**Reply by barbudor** [reply]
Baudrate setting is supposed to be saved to flash so it should be restored after boot.
However you may try to add it to your System#Boot rule:
```
Rule1 on System#Boot do Backlog Baudrate 9600; RuleTimer1 10 endon on Rules#Timer=1 do backlog SerialSend5 fe010f080000001c; RuleTimer1 10 endon
```

**Reply by miky-boy** [reply]
Hi,
Yes, with your rule works after reboot on 8.5.1 and any change on device attributes doesn't stop the readings. Thanks sir.
Why on 11.1 version stop? :(
Thanks and Regards
